staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -158,3 +164,44 @@ func compileRule(rule apiextensions.ValidationRule, env *cel.Env) (compilationRe
 func generateUniqueSelfTypeName() string {
 	return fmt.Sprintf("selfType%d", time.Now().Nanosecond())
 }
+
+type celEstimator struct {
+	root *celmodel.DeclType
+}
+
+func (c *celEstimator) EstimateSize(element checker.AstNode) *checker.SizeEstimate {
+	if len(element.Path()) == 0 {
+		// Path() can return an empty list, early exit if it does since we can't
+		// provide size estimates when that happens
+		return nil
+	}
+	currentNode := c.root
+	// cut off "self" from path, since we always start there
+	for _, name := range element.Path()[1:] {
+		var nextNode *celmodel.DeclType
+		switch name {
+		case "@items", "@values":
+			nextNode = currentNode.ElemType
+			if nextNode == nil {
+				return nil
+			}
+		case "@keys":
+			nextNode = currentNode.KeyType
```

Comment:
same question here, nextNode doesn't get assigned to currentNode anywhere... should it be?

Reply:
Fixed